### ISSUE TITLE
diffu-GRPO Training Time

### ISSUE THREAD

**Issue by Yeez-lee** [dllm-reasoning/d1#11]
Dear Author,

Thanks for open-sourcing these codes since LLaDa will not release training codes anymore. I wonder how long it would take to train LLaDa-8B with diffu-GRPO on 8 NVIDIA A100-80G GPU as mentioned in Sec. B.1? Thanks!

**Comment by siyan-zhao** [comment]
Hi, it took ~ 2 days to converge, depending on each task the time may vary.

**Comment by matrix-net** [comment]
FYI

3072/9995000 [12:23:03<7138:34:19,  2.57s/it]

6 48GB GPUs

**Comment by devaansh100** [comment]
Online RL is typically slow due to generation of the completions. For diffu-GRPO, this issue is further exacerbated due to the slower dLLM inference, and the few additional forward passes needed for prompt masking.

Which task are you training on, and what's your configuration file? 

Regardless, it's not surprising. For instance, the countdown dataset contains > 250k datapoints. Setting the config to then train for multiple epochs, combined with a low per-device batch size can give this training time. But you don't need to worry about that - in our experiments, we found the best checkpoints come in by ~10k steps. You could check our paper for more details.

**Comment by siyan-zhao** [comment]
Yes the total training time is huge because we set train epoch to 10, but we don’t need it to complete all epochs. It can converge earlier. You can early stop the experiments if the reward curve converges
